### PR TITLE
add a JIT

### PR DESCRIPTION
**Pull request by emm312**

### CODE REVIEW

**Comment by mauro-balades** [comment]
https://github.com/snowball-lang/snowball/pull/40/commits/d0393efc27c592014edaa20150180fbb600bd744#diff-63334f106abfc4439100cf86cd2b83bdbf88bf4a6b101f176ec26bad6d3ce328L2

File not found, fix this relative include

File: app/commands/run.h [hunk 1]
```diff
@@ -16,6 +16,7 @@ namespace snowball {
 namespace app {
 namespace commands {
 int run(app::Options::RunOptions p_opts) {
+    DUMP(p_opts.jit);
```
Comment (mauro-balades) [inline app/commands/run.h:19]: remove this before merge, thanks

File: src/constants.h [hunk 2]
```diff
@@ -85,7 +85,7 @@ typedef int32_t snowball_int_t;
 #define _SNOWBALL_LEXER_DEBUG    0
 #define _SNOWBALL_PARSER_DEBUG   0
 #define _SNOWBALL_CODEGEN_DEBUG  0
-#define _SNOWBALL_BYTECODE_DEBUG 1
+#define _SNOWBALL_BYTECODE_DEBUG 0
```
Comment (mauro-balades) [inline src/constants.h:88]: change it back to `1`

**Review by mauro-balades — CHANGES_REQUESTED** [review]
just some small changes. Didn't you send me that it segfaulted or smth? It needs to load runtime and other shared libs, idk how we could implement that, ill look into it.

File: build_scripts/dbg.sh [hunk 3]
```diff
@@ -1,6 +1,6 @@
 set -ex
 mkdir -p bin/Debug
-cmake -DCMAKE_INSTALL_PREFIX:PATH=/usr -G "Ninja" -DCMAKE_BUILD_TYPE=Debug \
+cmake -DCMAKE_INSTALL_PREFIX:PATH=/usr -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Debug \
```
[the remark this answers is not in the document]
Comment (emm312) [reply build_scripts/dbg.sh]: ninja doesn’t colour the errors

File: app/commands/run.h [hunk 4]
```diff
@@ -48,6 +48,7 @@ int run(app::Options::RunOptions p_opts) {
     compiler->compile(p_opts.silent);
     if (p_opts.jit) {
         compiler->run_jit();
+        return 0;
```
[the remark this answers is not in the document]
Comment (emm312) [reply app/commands/run.h:52]: you marked it as void in an example

File: app/commands/run.h (hunk 1 above)
Comment (emm312) [reply app/commands/run.h:19]: ok

File: src/constants.h (hunk 2 above)
Comment (emm312) [reply src/constants.h:88]: no

File: app/commands/run.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (mauro-balades) [reply app/commands/run.h:52]: It's OK, u can change it

**Pull request closed without merging**